lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -253,34 +278,68 @@ def _disable_if_needed(self, plugin: OpenSearchPlugin) -> bool:
         except KeyError as e:
             raise OpenSearchPluginMissingConfigError(e)
 
+    def _compute_settings(
+        self, config: OpenSearchPluginConfig
+    ) -> Tuple[Dict[str, str], Dict[str, str]]:
+        """Returns the current and the new configuration."""
+        current_settings = ClusterTopology.get_cluster_settings(
+            self._charm.opensearch,
+            include_defaults=True,
+        )
+        to_remove = dict(
+            zip(config.config_entries_to_del, [None] * len(config.config_entries_to_del))
+        )
+        new_conf = {
+            **current_settings,
+            **to_remove,
+            **config.config_entries_to_add,
+        }
```

Comment:
wouldn't this result everytime in the same dict per plugin?

Reply:
@Mehdi-Bendriss the last always takes precedence. That is how it can look:
```
>>> current_settings = {1:2, 3:4}
>>> to_remove = {3: None}
>>> to_add = {3: 5}
>>> print({**current_settings, **to_remove})
{1: 2, 3: None}
>>> print({**current_settings, **to_remove, **to_add})
{1: 2, 3: 5}
>>> print(dict(current_settings | to_remove | to_add))
{1: 2, 3: 5}
```

Reply:
I think this is my point. If we're always adding `to_add` after `to_remove` - why add `to_remove` at all?

Reply:
Not if they are trying to manipulate different keys